keras/engine/training.py

```diff
@@ -84,21 +85,36 @@ def compile(self, optimizer,
                 can specify them via the `target_tensors` argument. It can be
                 a single tensor (for a single-output model), a list of tensors,
                 or a dict mapping output names to target tensors.
+            check_array_lengths: By default, Keras will ensure that all input
+                and output tensors have a matching batch size for each batch.
+                If you would like to skip this check and allow variable batch
+                sizes among input/output tensors, this parameter can be set
+                to True.
```

Comment:
I would think `check_array_lengths` should be `False` to skip the check

Reply:
Yep --- you're right. I will update the doc

Reply:
Thanks for taking a look at this, BTW :)